app/models/concerns/order_filtering.rb

```diff
@@ -56,11 +56,19 @@ def priority
     end
 
     def due_after(time)
-      where('order_transports.scheduled_at >= ?', time).join_order_transports
+      where("CASE WHEN orders.detail_type='GoodCity' THEN
+                order_transports.scheduled_at >= (?)
+            ELSE
+                shipment_date >= (?)
+            END", time, time.to_date).join_order_transports
```

Comment:
what happens if more than one detail_type is selected?

Reply:
unrelated to comment above, but using heredoc `<<-SQL` would fit nicely here : )

Reply:
@steveyken, It works fine with more than one detail_type selected.